data-pipeline/src/trace_exporter.rs

```diff
@@ -176,25 +176,25 @@ impl TraceExporterBuilder {
     }
 
     pub fn set_tracer_version(&mut self, tracer_version: &str) -> &mut TraceExporterBuilder {
-        self.tracer_version = Some(String::from(tracer_version));
+        self.tracer_version = tracer_version.to_owned();
```

Comment:
I've always known `to_owned()` to be idiomatic for going from &str to String. Happy to revert this if it was done intentionally.

Reply:
It was done intentionally. The reasoning behind it is, at least in this case, the builder won't cross multiple function boundaries. It's called from an init function and it's never reused so it doesn't need to own all the resources. Therefore cloning all the values and then dropping them at the end of the function didn't seem a good choice in terms of memory management. If you foresee any use case in which the builder has to own all the resources go for it if not it's probably better to not allocate all that strings but that's my opinion.

Reply:
Nowadays, `String::From` and `to_owned()` are functionally equivalent, in both cases you're creating a String that the Builder owns. From a style perspective I personally prefer `to_owned()` as it looks cleaner (to me) and conveys intention of creating something the builder will own. 

If the intention is that the builder should only hold the references, then I can change it to do that. But, is that something we really want to do? Because at the end of the day we probably want `TracerTags` to not hold references as they will likely get used in another thread and the function that calls the builder may drop the references before we're done with them?

We could keep `TracerTags` members as `String`, but have the Builder use `&str` and do the clone in the `build()` function to defer the memory allocation as long as possible if we think it's worth it.

Reply:
I'm in favor of `to_owned` as well.